### PR TITLE
Add changelogger package

### PR DESCRIPTION
**Pull request by sinuscosinustan**
###### Description of changes

Changelogger generates and manages a changelog file in a project (written in Markdown). It can also generate Debian changelogs based on the content in the Markdown changelog.

GitHub repository: https://github.com/MarkusFreitag/changelogger

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/changelogger/default.nix
Comment (azahi) [inline pkgs/tools/misc/changelogger/default.nix:25]: Please specify the date here. It should be `1970-01-01T00:00:00` because the upstream uses gorelaser which follows this template.
```suggestion
    "-X github.com/MarkusFreitag/changelogger/cmd.BuildDate=1970-01-01T00:00:00"
```

File: pkgs/tools/misc/changelogger/default.nix
Comment (azahi) [inline pkgs/tools/misc/changelogger/default.nix:5]: This shouldn't be included as an input.
```suggestion
```

**Review by azahi — CHANGES_REQUESTED** [review]
Please amend your commit messages as it outlined in the [guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes). Specifically, the commit message should continue with lowercase after you have specified the scope, e.g. `init: package at 1.0.0` and not `init: Package at 1.0.0`. You are also not required to sign-off your commits.

**Comment by tomsiewert** [comment]
> Please amend your commit messages as it outlined in the [guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes). Specifically, the commit message should continue with lowercase after you have specified the scope, e.g. `init: package at 1.0.0` and not `init: Package at 1.0.0`. You are also not required to sign-off your commits.

Thank You for the review! I applied your suggestions, adapted the commit messages and removed the `sign-off`.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 179450` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>changelogger</li>
  </ul>
</details>

File: pkgs/tools/misc/changelogger/default.nix
Comment (azahi) [inline pkgs/tools/misc/changelogger/default.nix:41]: I doubt that Go can be built for all platforms. It would be safer to remove this line so it would default to Linux + Darwin.

**Review by azahi — CHANGES_REQUESTED** [review]
You still have an uppercase bit in your commit to the maintainer list.

**Review by azahi — APPROVED** [review]

File: pkgs/tools/misc/changelogger/default.nix
Comment (AndersonTorres) [inline pkgs/tools/misc/changelogger/default.nix:41]: No need for pretty tabulation here.

```suggestion
    description = "A tool to manage your changelog file in Markdown";
    homepage = "https://github.com/MarkusFreitag/changelogger";
    changelog = "https://github.com/MarkusFreitag/changelogger/blob/v${version}/CHANGELOG.md";
    license = licenses.mit;
    maintainers = with maintainers; [ tomsiewert ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**
